File: retos/estructurasDatos/matrices/agentes-secretos/AgentesSecretos.java

public class AgentesSecretos {
    public static void main(String[] args) {

        String str = "OSELNKTEPT-ANUEEREE";
        int filas = (int) Math.ceil(str.length() / 4.0);
        int col = 4;
        int index = 0;
        String mensaje = "";

      

        char matriz[][] = new char[filas][col];

        // Llenar la matriz por columnas (col -> filas)
        for (int j = 0; j < col; j++) {
            for (int i = 0; i < filas; i++) {
                matriz[i][j] = (index < str.length()) ? str.charAt(index++) : 'X';
            }
        }

        // Construir mensaje leyendo por filas
        for (int i = 0; i < filas; i++) {
            for (int j = 0; j < col; j++) {
                mensaje += matriz[i][j];
            }
        }
        System.out.println("*************** Mensaje desencriptado *********************");
        System.out.println("*                                                         *");
        System.out.println("*                                                         *");
        System.out.println("*          " + mensaje.replace("X", "") + "                            *");
        System.out.println("*                                                         *");
        System.out.println("***********************************************************");
      
    }
}

File: retos/estructurasDatos/matrices/agentes-secretos/agentes_secretos.py
import math

str = "OSELNKTEPT-ANUEEREE"
filas = math.ceil(len(str)/4)
col =4
matriz = [['' for _ in range(col)] for _ in range(filas)]
index =0
print(str[index])

for j in range(col):
    for i in range(filas):
            matriz[i][j] = str[index] if index < len(str) else  'X'
            index += 1
        
print(matriz)

    
mensaje = ''
for i in range(filas):
    for j in range(col):
            mensaje += matriz[i][j]
            
print(mensaje)
        